lib/galaxy/util/__init__.py

```diff
@@ -68,6 +68,9 @@
 FILENAME_VALID_CHARS = '.,^_-()[]0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
 
 
+defaultdict = collections.defaultdict
+
```

Comment:
I get the impulse here - it is a utility so add it to galaxy.util - but I'd generally prefer dependencies on ``collections`` over dependencies on ``galaxy.util`` in code that consume this. ``galaxy.util`` is very light and doesn't have many nested dependencies but it will never be a portable as ``collections``. I'm not -1 or anything - I'm going to merge this - just sharing my two cents.

Reply:
@jmchilton I would have imported from collections, but I'm using `defaultdict` in managers.jobs, but doing `from collections import defaultdict` doesn't work there because we also have managers.collections, this is a hack around that.